modules/@apostrophecms/ui/ui/apos/mixins/AposModifiedMixin.js

```diff
@@ -25,7 +25,8 @@ export default {
     // Returns true if the cancellation does occur
     async confirmAndCancel() {
       let dismiss;
-      if (this.isModified) {
+      const modified = typeof this.isModified === 'function' ? this.isModified() : this.isModified;
+      if (modified) {
```

Comment:
@ValJed Unrelated to main pr, but i noticed array's `isModified` is a function, not a boolean. The bug it was causing was that array modals always acted as if they had been modified and threw a confirm when exiting a clean modal.

Reply:
oh wow nicely spotted, wouldn't it be a better solution to use a computed everywhere or a function ?